File: pulumi/k3s-prod-1/networking/cert-manager.go
package networking

import (
	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/core/v1"
	"github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/helm/v3"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func NewCertManager(ctx *pulumi.Context, env string, namespace corev1.Namespace) error {
	var opts []pulumi.ResourceOption
	opts = append(opts, pulumi.DependsOn([]pulumi.Resource{namespace}))

	_, err := helm.NewChart(ctx, "cert-manager", helm.ChartArgs{
		Chart:     pulumi.String("cert-manager"),
		Version:   pulumi.String("corev1.14.2 "),
		Namespace: pulumi.String(namespace.Metadata.Name()),
		FetchArgs: helm.FetchArgs{
			Repo: pulumi.String("https://charts.jetstack.io"),
		},
		// https://cert-manager.io/docs/installation/helm/
		Values: pulumi.Map{},
	}, opts...)

	return err
}

File: pulumi/k3s-prod-1/networking/cert_manager.py
from pulumi import ResourceOptions
from pulumi_kubernetes.core.v1 import Namespace
from pulumi_kubernetes_cert_manager import CertManager, ReleaseArgs


class NewCertManager:
    NAMESPACE = "cert-manager"

    def __init__(self, provider):
        self.provider = provider

        self.ns = Namespace(
            "cert-manager",
            metadata={"name": self.NAMESPACE},
            opts=ResourceOptions(
                provider=self.provider,
            ),
        )

        # Install cert-manager into our cluster.
        self.manager = CertManager(
            "cert-manager",
            install_crds=True,
            helm_options=ReleaseArgs(
                namespace=self.NAMESPACE,
            ),
            opts=ResourceOptions(
                provider=self.provider,
            ),
        )
